File: docs/snippets/all/archetypes/graph_undirected.cpp
//! Log a simple undirected graph.

#include <rerun.hpp>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_graph_undirected");
    rec.spawn().exit_on_failure();

    rec.log(
        "simple",
        rerun::GraphNodes({"a", "b", "c"})
            .with_positions({{0.0, 100.0}, {-100.0, 0.0}, {100.0, 0.0}})
            .with_labels({"A", "B", "C"}),
        rerun::GraphEdges({{"a", "b"}, {"b", "c"}, {"c", "a"}})
            // Optional: graphs are undirected by default.
            .with_graph_type(rerun::components::GraphType::Undirected)
    );
}

File: docs/snippets/all/archetypes/graph_undirected.py
"""Log a simple undirected graph."""

import rerun as rr

rr.init("rerun_example_graph_undirected", spawn=True)

rr.log(
    "simple",
    rr.GraphNodes(
        node_ids=["a", "b", "c"], positions=[(0.0, 100.0), (-100.0, 0.0), (100.0, 0.0)], labels=["A", "B", "C"]
    ),
    rr.GraphEdges(
        edges=[("a", "b"), ("b", "c"), ("c", "a")],
        # Optional: graphs are undirected by default.
        graph_type="undirected",
    ),
)
